File: web-service/src/main/java/Registration/Department.java
package Registration;

import Files.UpFile;
import core.SEntity;

/**
 * Department can be Faculty of CS or English Department ...etc
 */
public class Department extends SEntity {

	/**
	 * parent used for major / minor
	 */
	protected Department parent;
	protected String name, description;
	protected UpFile LogoImg;
}

File: front-end/src/app/models/registration/department.ts
import { SEntity } from '../core/s-entity';

/**
 * Department can be Faculty of CS or English Department ...etc
 * @author Marc Wafik
 */
export class Department extends SEntity {

	name: string;
	description: string;

	constructor() {
		super();
		this.name = '';
		this.description = '';
	}

}